packages/runtime/container-runtime/src/runningSummarizer.ts

```diff
@@ -274,10 +276,22 @@ export class RunningSummarizer implements IDisposable {
             case MessageType.SummaryNack:
                 return false;
             default:
-                return true;
+                return isRuntimeMessage(op)
+                    || RunningSummarizer.nonRuntimeOpCanTriggerSummary(this.configuration, this.heuristicData);
         }
     }
 
+    private static nonRuntimeOpCanTriggerSummary(
+        config: ISummaryConfiguration,
+        heuristicData: ISummarizeHeuristicData,
+    ): boolean {
+        // eslint-disable-next-line max-len
+        const opsSinceLastAck = heuristicData.lastOpSequenceNumber - heuristicData.lastSuccessfulSummary.refSequenceNumber;
+        return config.state === "enabled"
+            && (config.nonRuntimeSummarizeThreshold === undefined
```

Comment:
Can it ever be undefined config.state == enabled ?

Reply:
State is always defined for this type

Reply:
sorry, I meant to ask whether config.nonRuntimeSummarizeThreshold == undefined could ever happen when config.state == enabled

Reply:
I made it an optional property. Should I switch it?

Reply:
Wouldn't that break the type tests? I think it's fine if it's optional for now